plans/destroy.pp

```diff
@@ -15,20 +15,50 @@
 #
 plan provision::destroy(
   Provision::CloudProvider  $provider          = 'aws',
-  String[1]                 $resource_name     = 'provision',
+  String[1]                 $resource_name,
   Optional[String[1]]       $region            = 'us-west-2',
-  Provision::ProviderOptions $provider_options = undef,
+  Optional[Provision::ProviderOptions] $provider_options = undef,
+  Optional[String[1]]       $profile            = undef,
+  Optional[String[1]]       $project            = undef,
+  Optional[String[1]]       $subnetwork_project = undef,
 ) {
   $_tf_dir = "terraform/${provider}/"
 
+  if $provider == 'gcp' {
+    $google_credentials = system::env('GOOGLE_CREDENTIALS')
```

Comment:
also this is most of code is common with create and destroy plan, if we can create method and move there and call that here?

Reply:
Unfortunately, plans don't support methods. We could possibly create a function.
@Ramesh7 This reminds me of AWS credentials: do we need to pass `profile` explicitly to the aws provider? If we leave it out, the environment variable will be taken automatically by the aws terraform provider.

Reply:
Yeah, the common code is converted to function and being called wherever required. @Ramesh7

Reply:
Yesterday I had discussion with @davidsandilands around using ENV variables and what we come to the conclusion is we can add that ENV support in next phase.
I am up for adding ENV support in current phase as well if we feel its critical?